### ISSUE TITLE
Experimental Feature Idea: autoGPT 

### ISSUE THREAD

**Issue by gipo355** [Bryley/neoai.nvim#14]
Just thinking out loud, I don't think there's anything yet that integrated autogpt inside nvim, correct me if wrong.
Would be amazing to give a general direction prompt, like a project scope prompt with objectives and let autoGPT create files and write code autonomously while researching the web too..

Would be an interesting experimental project imho.

Imagine allowing autogpt to operate in a specific directory to handle a branch for example.

could possibly be having autogpt installed in a specific path a requirement for the feat to be enabled?

**Comment by gipo355** [comment]
Yeah of course, this felt more like some kind of brainstorming.
It just hit me when using neoai as it would be cool to control autogpt from there.
It could be a separate plugin which requires neoai and an ENV with the path to the autoGPT script.
I know autogpt writes files to a .workspace_dir, don't know actually if it's possible to "integrate it" into a project in a different dir to allow it to write, modify files.

Sure it would be quite revolutionary to have an AI agent working on its own alongside the dev.

The difference would be that instead of waiting passively for input on what to do or change the old gpt style, it would just do its own thing and just require confirmations.

**simple prompt example:**
```
Name: FrontEndDeveloper-GPT
Description: You are in charge of developing the front-end website for the project(describe project) and write its code in the repo/frontend folder with nodejs npm and typescript. you can install popular npm libraries like react.
goal 1: website must be responsive
goal 2: website must follow best accessibility standards
goal 3: website must be following best practice guidelines for a modern website
goal 4: the design must be modern and have a dark theme variant
goal 5: the website will integrate with the backend code from the repo/backend folder
```
one can only dream

**Comment by nyngwang** [comment]
@gipo355 Lol, I don't that's possible with this simple project. What you're saying is carrying all those heavy stuff from https://github.com/Significant-Gravitas/Auto-GPT(more than 10k+ lines of code) to here and integrating it with Neovim APIs. That's not as simple as you might think. That's not realistic, to be honest.

On the other hand, if the purpose of Auto-GPT is to "do all things himself", then I don't think it's reasonable to integrate it into an "editor" like Neovim, which is designed to help "us/human" to write code efficiently. (Hope this doesn't hurt your feeling.)

**Comment by gipo355** [comment]
@nyngwang yeah indeed, guess you are right. Didn't mean it to sound "simple task", was thinking out loud of what I thought would be an insane feat at the time. It's probably better to just keep a terminal open on a second tab (or toggleterm) to control it as for letting AutoGPT work in a different subdir would require their repo to support it and introduce concerning security risks.

The vision I had was not about letting "autoGPT" make everything by itself but more of giving it a task like building a simple express API for the frontend you are generating handling the prompts from inside neovim where you later go and model the code it generated.

Maybe this can be closed I guess as a not pertinent issue, thank you for the feedback!
